quiche/src/lib.rs

```diff
@@ -3336,21 +3338,25 @@ impl Connection {
                 ecn_counts: None, // sending ECN is not supported at this time
             };
 
-            // ACK-only packets are not congestion controlled so ACKs must be
-            // bundled considering the buffer capacity only, and not
-            // the available cwnd.
-            if push_frame_to_pkt!(b, frames, frame, left) {
-                pkt_space.ack_elicited = false;
+            // When a PING frame needs to be sent, avoid sending the ACK if
+            // there is not enough cwnd available for both.
+            if !ack_elicit_required || frame.wire_len() < cwnd_available {
```

Comment:
should this be ACK frame wire length + PING frame wire length?

Reply:
Ok, updated.